### PR TITLE
[BugFix] fix precision loss issue when importing datetime type data

### LINKED ISSUE #26782: ms precision lost when importing datetime type data
### Steps to reproduce the behavior (Required)
1. CREATE TABLE 
```sql
CREATE TABLE test_datetime (
    c1 int,
    c2 datetime
) DUPLICATE KEY(c1)
DISTRIBUTED BY HASH(c1)
BUCKETS 3
PROPERTIES("replication_num" = "1");
```
2. INSERT INTO
```sql
insert into test_datetime values (1,'2020-01-01 00:00:00.0123');
```
3. SELECT 
```sql
select * from test_datetime;
```

### Expected behavior (Required)
```sql
mysql> select * from test_datetime;
+------+----------------------------+
| c1   | c2                         |
+------+----------------------------+
|    1 | 2020-01-01 00:00:00.012300 |
+------+----------------------------+
1 row in set (0.03 sec)
```
### Real behavior (Required)
```sql
mysql> select * from test_datetime;
+------+---------------------+
| c1   | c2                  |
+------+---------------------+
|    1 | 2020-01-01 00:00:00 |
+------+---------------------+
1 row in set (0.04 sec)
```
### StarRocks version (Required)
main 754a8f4f6ea7f3c44224dcb4ca02044b7a421fee

### PR DESCRIPTION
**Pull request by silverbullet233**
Fixes #26782

for INSERT query, FE converts datetime type data to string format and BE parsed data from this string, If the datetime type has milliseconds, we don't handle it correctly

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/util/stack_trace_mutex.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <array>
```
Comment (packy92) [inline be/src/util/stack_trace_mutex.cpp:15]: Is this necessary?

Comment (silverbullet233) [reply be/src/util/stack_trace_mutex.cpp:15]: > Is this necessary?

I met a compile error in my test env, this file uses std::array but there is no explicit include, I think add this include is reasonable

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/DateLiteral.java|1|1|100.00%|[]|

**Pull request merged**